test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,171 @@
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/test.h"
+
+namespace test_harness {
+/* Defines what data is written to the tracking table for use in custom validation. */
+class tracking_table_cache_resize : public test_harness::workload_tracking {
+
+    public:
+    tracking_table_cache_resize(
+      configuration *config, const bool use_compression, timestamp_manager &tsm)
+        : workload_tracking(config, use_compression, tsm)
+    {
+    }
+
+    void
+    set_tracking_cursor(const tracking_operation &operation, const uint64_t &, const std::string &,
+      const std::string &, wt_timestamp_t ts, scoped_cursor &op_track_cursor) override final
+    {
+        uint64_t cache_size =
+          ((WT_CONNECTION_IMPL *)connection_manager::instance().get_connection())->cache_size;
+        op_track_cursor->set_key(op_track_cursor.get(), ts);
+        op_track_cursor->set_value(op_track_cursor.get(), operation, cache_size);
+    }
+};
+
+/*
+ * This test continuously writes 2MB transactions into the database, while switching the connection
+ * cache size between 1MB and 500MB. When transactions are larger than the cache size they are
+ * rejected, so only transactions made when cache size is 500MB will be allowed.
+ */
+class cache_resize : public test_harness::test {
+    public:
+    cache_resize(const test_harness::test_args &args) : test(args)
+    {
+        init_tracking(new tracking_table_cache_resize(_config->get_subconfig(WORKLOAD_TRACKING),
+          _config->get_bool(COMPRESSION_ENABLED), *_timestamp_manager));
+    }
+
+    void
+    custom_operation(test_harness::thread_context *tc) override final
+    {
+        bool increase_cache = false;
+        const std::string small_cache_size("cache_size=1MB");
+        const std::string big_cache_size("cache_size=500MB");
+        uint64_t prev_cache_size;
+
+        while (tc->running()) {
+            tc->sleep();
+
+            WT_CONNECTION *conn = connection_manager::instance().get_connection();
+            prev_cache_size = ((WT_CONNECTION_IMPL *)conn)->cache_size;
+
+            testutil_check(conn->reconfigure(
+              conn, increase_cache ? big_cache_size.c_str() : small_cache_size.c_str()));
+            logger::log_msg(LOG_INFO,
+              "The cache size was updated from " + std::to_string(prev_cache_size) + " to " +
+                std::to_string(((WT_CONNECTION_IMPL *)conn)->cache_size));
+
+            /*
+             * We can only track the operation when the cache size is big enough otherwise it will
+             * be rolled back.
+             */
+            if (increase_cache) {
```

Comment:
Is this saying we only record cache increases to 500MB in the tracking table and not decreases to 1MB? We should be able to save both, otherwise in `validate` the `cache_size` variable we're asserting is > 500MB never changes and we won't fail if a successful write is made when `cache_size == 1MB`.

Reply:
The test was reworked as part of 069ae09dbbbc419a8ac2d101970c9b12f324292b, see the latest implementation as there are quite a few changes.